docs/source/user_guide/data_checks.ipynb

```diff
@@ -54,7 +54,150 @@
+    "\n",
+    "`ClassImbalanceDataCheck` checks if the target labels are imbalanced beyond a specified threshold for a certain number of CV folds. It returns errors for any classes that have less samples than double the number of CV folds, and warnings for any classes that fall below the set threshold percentage."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from evalml.data_checks import ClassImbalanceDataCheck\n",
+    "\n",
+    "X = pd.DataFrame([[1, 2, 0, 1],\n",
+    "                  [4, 1, 9, 0],\n",
+    "                  [4, 4, 8, 3],\n",
+    "                  [9, 2, 7, 1]])\n",
+    "y = pd.Series([0, 1, 1, 1, 1])\n",
+    "class_imbalance_check = ClassImbalanceDataCheck(threshold=0.25)\n",
+    "\n",
+    "for message in class_imbalance_check.validate(X, y):\n",
+    "    print (message.message)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Target Leakage\n",
+    "\n",
+    "Target leakage, also known as data leakage, can occur when you train your model on a dataset that includes information that should not be available at the time of prediction. This causes the model to score suspiciously well, but perform poorly in production. `TargetLeakageDataCheck` checks for features that could potentially be \"leaking\" information by calculating the Pearson correlation coefficient between each feature and the target to warn users if there are features are highly correlated with the target. Currently, only numerical features are considered."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from evalml.data_checks import TargetLeakageDataCheck\n",
+    "X = pd.DataFrame({'leak': [10, 42, 31, 51, 61],\n",
+    "'x': [42, 54, 12, 64, 12],\n",
+    "'y': [12, 5, 13, 74, 24]})\n",
+    "y = pd.Series([10, 42, 31, 51, 40])\n",
+    "\n",
+    "target_leakage_check = TargetLeakageDataCheck(pct_corr_threshold=0.8)\n",
+    "for message in target_leakage_check.validate(X, y):\n",
+    "    print (message.message)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Invalid Target Data\n",
+    "\n",
+    "The `InvalidTargetDataCheck` checks if the target data contains any missing or invalid values. Specifically:\n",
+    "\n",
+    "* if any of the target values are missing, an error is returned\n",
+    "* if the specified problem type is a binary classification problem but there is more or less than two unique values in the target, an error is returned\n",
+    "* if binary classification target classes are numeric values not equal to {0, 1}, an error is returned because it can cause unpredictable behavior when passed to pipelines"
```

Comment:
This is great

Is the last point still true? Now that we have the label encoder in the classification pipeline, I think this use case will work fine. If I'm right, can we file something to either delete this from `InvalidTargetDataCheck`, or change it to a warning instead of an error? Certainly, providing something which looks suspicious could indicate the user has provided incorrect data for some reason, which makes me think showing a warning is a good idea rather than just deleting.

Reply:
Hmmm, good point! I just filed #1422 to track investigating and updating the docs accordingly. For now, I'll merge as is with the documentation since that's the current behavior.